packages/@aws-cdk/aws-ecs/test/ec2/ec2-service.test.ts

```diff
@@ -1262,7 +1264,110 @@ describe('ec2 service', () => {
       // THEN
       expect(service.node.metadata[0].data).toEqual('taskDefinition and launchType are blanked out when using external deployment controller.');
       expect(service.node.metadata[1].data).toEqual('Deployment circuit breaker requires the ECS deployment controller.');
+    });
+
+    test('add alarm config while service is constructed if deploymentAlarms config is specified with no behavior', () => {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const vpc = new ec2.Vpc(stack, 'MyVpc', {});
+      const cluster = new ecs.Cluster(stack, 'EcsCluster', { vpc });
+      addDefaultCapacityProvider(cluster, stack, vpc);
+      const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'Ec2TaskDef');
+
+      taskDefinition.addContainer('web', {
+        image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
+        memoryLimitMiB: 512,
+      });
+
+      const myAlarm = cloudwatch.Alarm.fromAlarmArn(stack, 'myAlarm', 'arn:aws:cloudwatch:us-east-1:1234567890:alarm:alarm1');
+
+      const service = new ecs.Ec2Service(stack, 'Ec2Service', {
+        cluster,
+        taskDefinition,
+        deploymentController: {
+          type: DeploymentControllerType.ECS,
+        },
+        deploymentAlarms: {
+          alarms: [myAlarm],
+        },
+      });
+
+      // THEN
+      const config = (service.node.defaultChild as CfnService).deploymentConfiguration as CfnService.DeploymentConfigurationProperty;
```

Comment:
i'm curious why those `as` are needed - especially there is another one below. is there a way to avoid using `as` ?

Reply:
I think we should use the `Template.fromStack()` matcher pattern that's found elsewhere in the test files: 

```typescript
const template = Template.fromStack(stack);
template.hasResourceProperties('AWS::ECS::Service', {
  DeploymentConfiguration: {
    Alarms: {
      ... // Your desired configuration here
    },
  },
});
```

Reply:
You can use the CFN spec to find the right field names: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-deploymentconfiguration.html

Reply:
Ya we can use that as austin said, Thank you austin
 const template = Template.fromStack(stack);
      template.hasResourceProperties('AWS::ECS::Service', {
        DeploymentConfiguration: {
          Alarms: {
            Enable: true,
            Rollback: true,
            AlarmNames: [myAlarm.alarmName],
          },
        },
      });

Reply:
I updated the test cases by using Template.fromStack() matcher pattern